ctfbot/cog.py

```diff
@@ -147,3 +167,181 @@ async def reminder(self, ctx: discord.ApplicationContext):
         else:
             await ctx.respond('Added reminder for this event')
             del data.reminders[ctx.channel_id]
+    
+
+    @commands.slash_command()
+    async def challenge(self, ctx: discord.ApplicationContext, chal_category: discord.Option(str), chal_name: discord.Option(str)):
+        banned_strings = ['→', '**', '~~', '@']
+        if any(banned_string in chal_category or banned_string in chal_name for banned_string in banned_strings):
+            await ctx.respond('Invalid character in challenge name/category')
+            return
+        guild: discord.Guild = ctx.guild
+        data = self.data.servers[ctx.guild_id]
+        for event_id, category_id in data.events.items():
+            category = guild.get_channel(category_id)
+            if category and ctx.channel_id == category.channels[3].id:
+                break
+        else:
+            await ctx.respond('This is not a CTF channel')
+            return
+        channel_challenges: discord.TextChannel = category.channels[2]
+        channel_general: discord.TextChannel = category.channels[3]
+        message_challenges: discord.Message = await channel_challenges.fetch_message(channel_challenges.last_message_id)
+        embed = message_challenges.embeds[0]
+        thread = None
+        for index, field in enumerate(embed.fields):
+            if field.name == f'**{chal_category}**':
+                for i in range(index + 1, len(embed.fields)):
+                    if embed.fields[i].value.startswith(chal_name):
+                        await ctx.respond('Challenge already exists')
+                        return
+                    elif embed.fields[i].name.startswith('**'):
+                        thread = await channel_general.create_thread(name=chal_category + '/' + chal_name, type=discord.ChannelType.public_thread)
+                        embed.insert_field_at(i, name='', value=chal_name + ' → ' + thread.mention, inline=False)
+                        break
+                else:
+                    thread = await channel_general.create_thread(name=chal_category + '/' + chal_name, type=discord.ChannelType.public_thread)
+                    embed.add_field(name='', value=chal_name + ' → ' + thread.mention, inline=False)
+                break
```

Comment:
Way to many levels of indention, should probably think about breaking parts into methods.

Reply:
cleaned up function